### PR TITLE
[feat] Improve canvas zoom #44

### PR DESCRIPTION
**Pull request by addegbenga**
- [x] issue #https://github.com/keep-starknet-strange/art-peace/issues/44
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

This PR enhances the zooming functionality of the canvas by focusing on zooming in on the area around the mouse cursor, rather than zooming the entire canvas.

### CODE REVIEW

File: frontend/src/canvas/Canvas.js [hunk 1]
```diff
@@ -35,58 +28,33 @@ const Canvas = props => {
...
-      setDragStartY(e.clientY)
-    }
+  const zoomHandler = (event) => {
+    const ele = canvasScaleRef.current
+    const {
+      k: newScale,
+      x: newCanvasPositionX,
+      y: newCanvasPositionY,
+    } = event.transform;
+    const transformValue = `translate(${newCanvasPositionX}px, ${newCanvasPositionY}px) scale(${newScale})`
```
Comment (b-j-roberts) [inline frontend/src/canvas/Canvas.js:54]: Can you set & check against min & max zoom scales? Hardcoded minScale = 1 & maxScale = 40 for now is good.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
I didn't know about the `d3` library, nice find! Exactly what I was looking for!

This seemed to patch an issue with positioning when opening dev tools as well.

File: frontend/src/canvas/Canvas.js (hunk 1 above)
Comment (addegbenga) [reply frontend/src/canvas/Canvas.js:54]: Okay, i will make the requested changes.

Comment (addegbenga) [reply frontend/src/canvas/Canvas.js:59]: Added the requested changes kindly review

**Review by b-j-roberts — APPROVED** [review]

**Comment by b-j-roberts** [comment]
Great work, thank you!

**Pull request merged**